src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -133,6 +133,46 @@ def _is_prosumer_node(area):
     return isinstance(area.strategy, StorageStrategy)
 
 
+def _accumulate_storage_trade(storage, area, accumulated_trades, past_market_types):
+    if storage.name not in accumulated_trades:
+        accumulated_trades[storage.name] = {
+            "type": "Storage",
+            "produced": 0.0,
+            "earned": 0.0,
+            "consumedFrom": {},
+            "spentTo": {},
+        }
+
+    markets = getattr(area, past_market_types)
+    if markets is None:
+        return accumulated_trades
+    else:
+        if type(markets) != list:
+            markets = [markets]
+        for market in markets:
+            for trade in market.trades:
+                if trade.buyer == storage.name:
+                    sell_id = area_name_from_area_or_iaa_name(trade.seller)
+                    accumulated_trades[storage.name]["consumedFrom"] = add_or_create_key(
+                        accumulated_trades[storage.name]["consumedFrom"],
+                        sell_id, trade.offer.energy)
+                    accumulated_trades[storage.name]["spentTo"] = add_or_create_key(
+                        accumulated_trades[storage.name]["spentTo"], sell_id, trade.offer.price)
+                elif trade.offer.seller == storage.name:
+                    accumulated_trades[storage.name]["produced"] -= trade.offer.energy
+                    # buyer = area_name_from_area_or_iaa_name(trade.buyer)
```

Comment:
Any reason for removing the price statistics here and below?

Reply:
It wasn't there in first place either, was neither a requirement if I have understood correctly.
![20190528_154149](https://user-images.githubusercontent.com/33482427/58555985-31a2ee00-821b-11e9-9a62-c0854e8cf561.jpg)
I was paying around while testing, I will remove it